wolfcrypt/src/port/liboqs/liboqs.c

```diff
@@ -50,9 +51,24 @@ static int liboqs_init = 0;
 
 static void wolfSSL_liboqsGetRandomData(uint8_t* buffer, size_t numOfBytes)
 {
-    int ret = wc_RNG_GenerateBlock(liboqsCurrentRNG, buffer, (word32)numOfBytes);
-    if (ret != 0) {
-        // ToDo: liboqs exits programm if RNG fails, not sure what to do here
+    int ret;
+    word32 numOfBytes_word32;
+
+    while (numOfBytes > 0) {
+        numOfBytes_word32 = (word32)numOfBytes;
+        numOfBytes -= numOfBytes_word32;
+        ret = wc_RNG_GenerateBlock(liboqsCurrentRNG, buffer,
+                                   numOfBytes_word32);
+        if (ret != 0) {
+            /* ToDo: liboqs exits programm if RNG fails,
```

Comment:
You can do a limited number of retries, say 10.
If still fails, return failure and allow liboqs to exit.
We can't stop there bad behaviour but it is kinda fatal at this point.
Customers can fix liboqs if they don't like it.

Reply:
Can't "return failure" -- liboqs in its infinite wisdom made the callback a `void`.  So it's either silently fail, or `abort()`, or something really complicated using a thread-local variable, which seems way out of scope for our purposes here.

Should we really bother retrying `wc_RNG_GenerateBlock()`?  Particularly without a `usleep` before each retry?  I guess you're thinking of `DRBG_NEED_RESEED`, but `wc_GenerateSeed()` uses `/dev/urandom` in the default and `WOLFSSL_GETRANDOM` implementations, so if they fail once they'll fail forever.

Reply:
I agree, not really worth doing anything complicated here.  However, can we remove the todo?  Nothing left to do.  It really is appropriate action.